packages/tools/devtools/devtools/src/index.ts

```diff
@@ -187,6 +192,17 @@ function mapContainerProps(
 	};
 }
 
+/**
+ * Attempts to retrieve the global Fluid Devtools instance, initializing it if it's not exists.
+ * @beta
+ */
+export function tryGetIFluidDevtools(logger?: ITelemetryBaseLogger): IFluidDevtools {
+	if (globalThis.IFluidDevtools === undefined) {
+		globalThis.IFluidDevtools = initializeDevtools({ logger });
+	}
+	return globalThis.IFluidDevtools;
```

Comment:
Hmm this would be creating a different instance of Devtools than the one a consumer gets by calling `initializeDevtools()` themselves. If we go this route then I think we should get rid of `initializeDevtools()`? We currently don't really enforce a singleton instance, each call to that function returns a separate one, it's just that applications are only supposed to call the function once, so they end up doing everything on the same instance. This would make it into a real singleton. Can't think of an obvious problem with not allowing multiple instances to exist, but would like to hear from the rest of the team.

Reply:
I would suggest instead modeling this to return `undefined` if no devtools have been initialized via `initializeDevtools`. It's up to the consumer then to decide if this returning `undefined` is an error or not.

Reply:
Also, there is already a singleton we can leverage, so we shouldn't need to touch `globalThis`.
https://github.com/microsoft/FluidFramework/blob/main/packages/tools/devtools/devtools-core/src/FluidDevtools.ts#L288

So you should be able to replace all of the logic inside this function with:
```suggestion
	return FluidDevtools.tryGet();
```

Reply:
Ah ha. How convenient!

Reply:
I do not see a harm in supporting singleton pattern for devtools, and in fact I believe this is preferred since I can't think of a usecase of storing multiple devtools instance for the lifecycle of an application. 

That said, as @alexvy86 mentioned, we should consolidate `tryGetIFluidDevtools()` and `initializeDevtools()` into a single method to avoid confusion, potential bugs, and unnecessary complexity in the API.

Reply:
> We currently don't really enforce a singleton instance, each call to that function returns a separate one, it's just that applications are only supposed to call the function once, so they end up doing everything on the same instance. This would make it into a real singleton. Can't think of an obvious problem with not allowing multiple instances to exist, but would like to hear from the rest of the team.

This isn't actually true. Under the hood, we _do_ already leverage a singleton. If `initializeDevtools` is called twice, we log a warning to the console and just return the existing one.

Reply:
> That said, as @alexvy86 mentioned, we should consolidate tryGetIFluidDevtools() and initializeDevtools() into a single method to avoid confusion, potential bugs, and unnecessary complexity in the API.

Note that `initializeDevtools` already returns the initialized instance. So if you need a reference to it immediately, you can just use what the function returns. This new function is for scenarios where it is assumed that an earlier stage in the app life cycle would have already initialized the devtools (as appropriate), and we want to do something more with it.